src/Fantomas.Core/ASTTransformer.fs

```diff
@@ -78,6 +78,16 @@ let mkSynAccess (vis: SynAccess option) =
     | Some(SynAccess.Private range) -> Some(stn "private" range)
     | Some(SynAccess.Public range) -> Some(stn "public" range)
 
+let (|AccessSynValSigAccess|) (valSigAccess: SynValSigAccess) =
+    match valSigAccess with
+    | SynValSigAccess.GetSet(accessibility = vis)
+    | SynValSigAccess.Single(accessibility = vis) -> vis
+
+let (|SynValSigAccessAll|) (valSigAccess: SynValSigAccess) =
+    match valSigAccess with
+    | SynValSigAccess.Single(vis) -> vis, None, None
+    | SynValSigAccess.GetSet(vis, g, s) -> vis, g, s
+
```

Comment:
Nit:
As they are so close to each other, I'd go with the same order in the APs: `GetSet`, then `Single`